pkg/sync/syncer.go

```diff
@@ -1032,3 +1040,34 @@ func NewSyncer(ctx context.Context, c types.ConnectorClient, opts ...SyncOpt) (S
 
 	return s, nil
 }
+
+// This method checks if the principal resource already exists. If it does not, we should put the resource.
+// The return value indicates if the resource was put or not by the boolean value.
+func (s *syncer) ensureResourceExistence(ctx context.Context, principal *v2.Resource) error {
```

Comment:
Probably better to call this `ensurePrincipalExists` or something like that, since we're not using it for all resources(e.g. we're not JITing the resource on the entitlement)

Reply:
Fixed in latest commit